src/Misc/layoutbin/runsvc.sh

```diff
@@ -9,4 +12,8 @@ fi
 # insert anything to setup env when running as a service
 
 # run the host process which keep the listener alive
-./externals/node/bin/node ./bin/AgentService.js
+./externals/node/bin/node ./bin/AgentService.js &
+PID=$!
+wait $PID
+trap - TERM INT
+wait $PID
```

Comment:
this logic (including why we wait twice) is explained here http://veithen.github.io/2014/11/16/sigterm-propagation.html